### PR TITLE
Add support to read/write space-padded fixed-length strings

### PR DESCRIPTION
**Pull request by vmsh0**
Following the discussion in issue #375, I'm offering this PR with the modifications we are using internally to support the datatype.

A recap of what this is, adapted from the commit message:

These strings are not part of the official s7comm specification, but
they are commonly found in real-world systems. They simply consist of a
byte array that contains characters forming a string, padded on the
right with spaces (ascii 32) when the string length is shorter than the
array.

This commit adds support for reading and writing, as well as usage in
tables (snap7.util.DB), and unit tests.

### CODE REVIEW

**Comment by vmsh0** [comment]
I kept this separate from #377 in case you only want the fix (and not this feature), but if you wish I can both changes on your last master and offer a single pull request with both things.

File: snap7/util.py [hunk 1]
```diff
@@ -410,6 +410,44 @@ def get_real(bytearray_: bytearray, byte_index: int) -> float:
...
+        or 'value' contains non-ascii characters.
+
+    Examples:
+        >>> data = bytearray(20)
+        >>> snap7.util.set_fstring(data, 0, "hello world", 15)
+        >>> data
+            bytearray(b'hello world    \x00\x00\x00\x00\x00')
+    """
+    if not isinstance(value, str):
+        raise TypeError(f"Value value:{value} is not from Type string")
```
Comment (gijzelaerr) [inline snap7/util.py:434]: to be consistent, I think validating the type should be handled by a type checker like mypy, and not be done manually.

**Review by gijzelaerr — CHANGES_REQUESTED** [review]
i don't think this affects any existing usage, so I don't object adding this to the lib. one small comment about type checking.

**Comment by vmsh0** [comment]
I have made the requested change as well as fixed the reported linting issues.

**Pull request merged**